src/staticdata.c

```diff
@@ -3454,8 +3456,6 @@ static void jl_restore_system_image_from_stream_(ios_t *f, jl_image_t *image, jl
             jl_code_instance_t *ci = (jl_code_instance_t*)obj;
             assert(s.incremental);
             ci->min_world = world;
-            if (ci->max_world != 0)
```

Comment:
This is really the key part of this change. Everything else is largely cosmetic.

Reply:
So if this was before implemented as `if (ci->max_world != 0 && !in(ci, new_specializations)) abort()` it would never have failed, but was some relic of some past thing?

Reply:
No, there two kinds of code instances like that:
1. Internal ones that require validation (max_world == 1), which are handled by the extra cache walk you commented on below
2. Internal ones that don't (max_world == `(size_t)-1`), which don't need any extra handling.

Reply:
okay, yeah, I guess that does make sense.

alright, SGTM/LGTM